server/starters/web-starter/src/main/java/com/oceanbase/odc/service/iam/auth/ldap/LdapUserDetailsContextMapper.java

```diff
@@ -35,6 +36,7 @@
 
 @Component
 @Slf4j
+@ConditionalOnProperty(value = {"odc.iam.auth.type"}, havingValue = "local")
```

Comment:
This was fixed by @ungreat , you could merge dev/4.2.x and push your branch again